### PR TITLE
Fix workload remove

### PR DESCRIPTION
**Pull request by hzxuzhonghu**
**What type of PR is this?**

**What this PR does / why we need it**:

**Which issue(s) this PR fixes**:
Fixes #

**Special notes for your reviewer**:

**Does this PR introduce a user-facing change?**:

```release-note

```

### CODE REVIEW

**Comment by hzxuzhonghu** [comment]
fix https://github.com/kmesh-net/kmesh/issues/154

File: pkg/controller/workload/workload_event.go [hunk 1]
```diff
@@ -108,19 +108,6 @@ func (svc *ServiceEvent) processWorkloadResponse(rsp *service_discovery_v3.Delta
 			err = fmt.Errorf("unsupport type url %s", rsp.GetTypeUrl())
 		}
 	}
-
```
Comment (supercharge-xsy) [inline pkg/controller/workload/workload_event.go:111]: is there a situation we need to consider？  rsp.GetResources() == nil & rsp.RemovedResources != nil

File: pkg/controller/workload/workload_event.go [hunk 2]
```diff
@@ -108,19 +108,6 @@ func (svc *ServiceEvent) processWorkloadResponse(rsp *service_discovery_v3.Delta
...
-	if rsp.RemovedResources != nil {
```
Comment (supercharge-xsy) [inline pkg/controller/workload/workload_event.go:112]: is there a situation we need to consider？ rsp.GetResources() == nil & rsp.RemovedResources != nil

Comment (hzxuzhonghu) [reply pkg/controller/workload/workload_event.go:112]: I have checked, in the function we use for range, if this is nil, it does not matter. 

Just we need to guarantee all the functions removeXXX or handleXX is safe with nil input

Comment (supercharge-xsy) [reply pkg/controller/workload/workload_event.go:112]: we need also change this condition to enter handlexx function， when rsp.GetResources() == nil  but rsp.RemovedResources != nil, should we ? line 101 :
`	if rsp.GetResources() != nil {
		switch rsp.GetTypeUrl() {
		case AddressType:
			err = handleAddressTypeResponse(rsp)
		case AuthorizationType:
			err = svc.handleAuthorizationTypeResponse(rsp)
		default:
			err = fmt.Errorf("unsupport type url %s", rsp.GetTypeUrl())
		}
	}`

Comment (hzxuzhonghu) [reply pkg/controller/workload/workload_event.go:112]: Not sure i can get what you mean

Comment (supercharge-xsy) [reply pkg/controller/workload/workload_event.go:112]: > Not sure i can get what you mean

in processWorkloadResponse function ，The current code implementation misses a situation ：when rsp has RemovedResources field without Resources field

**Comment by hzxuzhonghu** [comment]
@nlgwcy can you approve

**Comment by hzxuzhonghu** [comment]
@nlgwcy can you approve

**Pull request merged**